CMakeLists.txt

```diff
@@ -123,17 +123,32 @@ endif(BUILD_DOCKERAPP)
 if(BUILD_P11)
     find_package(LibP11 REQUIRED)
     add_definitions(-DBUILD_P11)
-endif(BUILD_P11)
 
-# Setup PKCS11
-if(TEST_PKCS11_MODULE_PATH)
-    add_definitions(-DTEST_PKCS11_MODULE_PATH="${TEST_PKCS11_MODULE_PATH}"
-                    -DTEST_PKCS11_ENGINE_PATH="${TEST_PKCS11_ENGINE_PATH}")
-endif(TEST_PKCS11_MODULE_PATH)
+    if (NOT PKCS11_ENGINE_PATH)
+        if (CMAKE_CROSSCOMPILING)
+            message(FATAL_ERROR "Please define the PKCS11_ENGINE_PATH on your target system")
+        endif()
+        foreach(engine IN ITEMS "/usr/lib/engines-1.1/pkcs11.so"
+                "/usr/lib/engines/pkcs11.so"
+                "/usr/lib/x86_64-linux-gnu/engines-1.1/pkcs11.so")
+
+            if(EXISTS ${engine})
+                message(STATUS "Detected pkcs11 engine path: ${engine}")
+                set(PKCS11_ENGINE_PATH ${engine})
+            endif()
+
+        endforeach()
+
+        if (NOT PKCS11_ENGINE_PATH)
+            message(FATAL_ERROR "Could not auto-detect path of PKCS11 engine, please specify PKCS11_ENGINE_PATH")
+        endif()
+
+        set(PKCS11_ENGINE_PATH "${PKCS11_ENGINE_PATH}" CACHE STRING "Credentials.zip for tests involving the server")
+    endif()
+endif(BUILD_P11)
 
 if(BUILD_SOTA_TOOLS)
     find_package(GLIB2 REQUIRED)
-    find_program(STRACE NAMES strace)
```

Comment:
I thought I left a comment here but apparently not. `strace` is still used by the `garage-push-cacert-used` test. (I don't think that test actually runs in docker, though, because we never install strace.) Otherwise this PR looks fine.

Reply:
Yes, I missed that somehow...

I'd suggest we'd either nuke this test or make sure it's executed on CI. Any preference?

Reply:
Seems easy to just put it into CI; that'd be my preference. It is testing something otherwise uncovered.